src/replication/log_consumer.cpp

```diff
@@ -187,8 +191,13 @@ namespace cubreplication
 		/* wait for all started tasks to finish */
 		er_log_debug_replication (ARG_FILE_LINE, "dispatch_daemon_task wait for all working tasks to finish\n");
 
-		m_lc.wait_for_tasks ();
+		m_prev_group_stream_position = m_curr_group_stream_position;
+		m_curr_group_stream_position = se->get_stream_entry_start_position ();
+
+		/* We need to wait for previous group to complete. Otherwise, we mix transactions from previous and current groups. */
+		m_p_dispatch_consumer->wait_for_complete_stream_position (m_prev_group_stream_position);
 
+		count_expected_transaction = 0;
```

Comment:
Flags are inherited. Probably were 3 reasons, for using flags:
- simplicity
- keeping flags may help us at debug
- GC not considered

I just use the old behavior to add slave GC.

Removing flags is optional. If adding flags affects something (performance for instance or creates confusion) it may be removed. Otherwise, we may add comments.

In any case, keeping or not the flags is not the purpose of current patch.

Reply:
The safest and probably final solution would be to use group complete info. The discussed design was to completely replace individual logging (for both transactional log and HA replication) of commit/abort and use only grouped logging. Do you want to challenge and change this?

If yes, the question of safety remains - do individual flags match group complete? 
If no, it is better to not rely on those flags even if they will be removed later.

Reply:
Once again, removing flags now is optional (all commits/abort should be in group commit info). The flags helps debug to match commits/abort in GC.

This was the FASTEST way for current patch, where we care about correctness, not necessary improving existing code . If we decide to remove the flags, we need to read finalized transactions from GC stream. Otherwise, assert to check expecting transactions is enough.